src/main/integration/java/com.juniorstart.juniorstart/Service/UserProfileTest.java

```diff
@@ -0,0 +1,127 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.model.*;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import java.util.*;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/** Represents an user service.
+ * @author Rafał Maduzia
+ * @version 1.0
+ */
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace= AutoConfigureTestDatabase.Replace.NONE)
+public class UserProfileTest {
+
+    @SpyBean
+    private UserProfileService userProfileService;
+
+    @Autowired
+    private UserDao userRepository;
+
+    @Autowired
+    private UserProfileRepository userProfileRepository;
+
+    List<String> technologyList  = new ArrayList<>();
+    List<String> userRoleList  = new ArrayList<>();
+
+    @Before
+    public void setUp() {
+        User user;
+        user = User.builder()
+                .name("Test")
+                .email("test2@test.com")
+                .emailVerified(true)
+                .password("Test%123")
+                .provider(AuthProvider.google).build();
+        userRepository.save(user);
+
+        UserTechnology userTechnology = new UserTechnology();
+        userTechnology.setTechnologyName("Java");
+
+        UserProfile userProfile = new UserProfile();
+        userProfile.setUser(user);
+        userProfile.setUserRole(UserRole.MENTOR);
+        userProfile.addUserTechnology(userTechnology);
+        userProfileRepository.save(userProfile);
+    }
+
+    @Test
+    public void Should_FindUserProfileByRoleAndTechnology() {
+        technologyList.add("java");
```

Comment:
Pownno się użyć biblioteki Mockito do takich testów. Zobacz testy w UserServiceTest

Reply:
Nie przejdę one przez CI

Reply:
Co rozumiesz, że nie przejdą przez CI ? Hmmm rozmawiałem w tej sprawie bodajże z Adrianem i mówił , żebym zostawił. Tylko teraz pytanie w takim razie, czy mam to przepisywać na nowo na Mocki ?

Reply:
Przyjeliśmy konwencje, że powiniśmy pisać testy przy użyciu Mocków, ponieważ nie przechodzą testy na CI. Na CI nie ma dostępu do bazy danych i przez to testy nie przechodzą.